artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -95,13 +95,17 @@ func consumeAllDelayFiles(base phaseBase) error {
 		return err
 	}
 	delayFunctions := getDelayUploadComparisonFunctions(base.repoSummary.PackageType)
-	if len(filesToConsume) > 0 && len(delayFunctions) > 0 {
-		log.Info("Starting to handle delayed artifacts uploads...")
-		if err = handleDelayedArtifactsFiles(filesToConsume, base, delayFunctions[1:]); err == nil {
-			log.Info("Done handling delayed artifacts uploads.")
-		}
+	if len(filesToConsume) == 0 || len(delayFunctions) == 0 {
+		return nil
+	}
+
+	log.Info("Starting to handle delayed artifacts uploads...")
+	if err = handleDelayedArtifactsFiles(filesToConsume, base, delayFunctions[1:]); err != nil {
+		return err
 	}
-	return err
+
+	log.Info("Done handling delayed artifacts uploads.")
+	return deleteAllFiles(filesToConsume)
 }
```

Comment:
Let's add a comment explaining why the first element is skipped in `delayFunctions[1:]` in the following line in inside this function.
```
if err = handleDelayedArtifactsFiles(filesToConsume, base, delayFunctions[1:]); err != nil {
```